src/test/java/jenkins/plugins/openstack/compute/internal/OpenstackTest.java

```diff
@@ -4,29 +4,301 @@
 import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.equalTo;
 import static org.junit.Assert.fail;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyMapOf;
+import static org.mockito.Matchers.argThat;
 import static org.mockito.Mockito.*;
 
 import org.hamcrest.Matchers;
+import org.junit.Before;
 import org.junit.Test;
 import org.jvnet.hudson.test.Issue;
+import org.mockito.Answers;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
 import org.mockito.invocation.InvocationOnMock;
 import org.mockito.stubbing.Answer;
 import org.openstack4j.api.OSClient;
 import org.openstack4j.api.compute.ComputeFloatingIPService;
+import org.openstack4j.api.compute.ext.ZoneService;
+import org.openstack4j.api.image.ImageService;
+import org.openstack4j.api.storage.BlockVolumeSnapshotService;
 import org.openstack4j.api.exceptions.ClientResponseException;
 import org.openstack4j.model.common.ActionResponse;
 import org.openstack4j.model.compute.Fault;
 import org.openstack4j.model.compute.FloatingIP;
 import org.openstack4j.model.compute.Server;
 import org.openstack4j.model.compute.builder.ServerCreateBuilder;
+import org.openstack4j.model.compute.ext.AvailabilityZone;
+import org.openstack4j.model.image.Image;
+import org.openstack4j.model.storage.block.Volume;
+import org.openstack4j.model.storage.block.VolumeSnapshot;
 import org.openstack4j.openstack.compute.domain.NovaFloatingIP;
 
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
 import java.util.Iterator;
 import java.util.List;
+import java.util.Map;
 
+@SuppressWarnings({
+        "rawtypes",
+        "unchecked"
+})
 public class OpenstackTest {
+    private @Mock Image mockImageWithNullName;
+    private @Mock Image mockImageNamedFoo;
+    private @Mock Image mockImageNamedBar1;
+    private @Mock Image mockImageNamedBar2;
+    private @Mock Image mockImageNamedBar3;
+    private @Mock VolumeSnapshot mockVolumeSnapshotWithNullName;
+    private @Mock VolumeSnapshot mockVolumeSnapshotNamedFoo;
+    private @Mock VolumeSnapshot mockVolumeSnapshotNamedBar1;
+    private @Mock VolumeSnapshot mockVolumeSnapshotNamedBar2;
+    private @Mock VolumeSnapshot mockVolumeSnapshotNamedBar3;
+    private @Mock AvailabilityZone mockAZ1;
+    private @Mock AvailabilityZone mockAZ2;
+    private @Mock AvailabilityZone mockAZ3;
+
+    private @Mock ImageService mockIS;
+    private @Mock BlockVolumeSnapshotService mockBVSS;
+    private @Mock ZoneService mockZS;
+    private @Mock(answer = Answers.RETURNS_DEEP_STUBS) OSClient mockClient;
```

Comment:
OK, I'll update the code to create these each each method.

Personally I prefer sharing simple mocks like this, especially as `Foo mockFoo = mock(Foo.class);` starts to get really messy if Foo is Foo<Bar> instead - you can't do `Foo<Bar> mockFoo = mock(Foo<Bar>.class);` whereas `@Mock` doesn't cause warnings when using generic types.  Everyone has their own preferences ;-)